src/librbd/io/ImageRequestWQ.cc

```diff
@@ -423,6 +442,23 @@ void *ImageRequestWQ::_void_dequeue() {
       }
     } else if (m_require_lock_on_read) {
       return nullptr;
+    } else {
+      if (m_image_ctx.max_read_iops) {
```

Comment:
I will move the `if (m_require_lock_on_read)` and `if (m_image_ctx.max_read_iops)` under the same block scope.